lib/book.rb

```diff
@@ -0,0 +1,17 @@
+require './rental'
+
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def rent(date, person)
+    new_rental = Rental.new(date, person, self)
+    @rental << new_rental
+  end
```

Comment:
- [ ] Awesome job 👏🏻 , `associations` in ruby are exciting and understanding it will help you to be a better Ruby on Rails developer in the upcoming time. But it seems like you still need some clarity on association. The book has many rentals, which means it does contain an array of `rentals` which you did very well but adding the rentals in your function using **<<** is not going to help you because every time a person adds a new book then a new copy of an array will be created. Instead, you can add this `@rental << new_rental` you can add the line in your `Rental.rb` file like `book.rentals << self`.  Now the doubt is how it will work then you need to understand Ruby is a language of convention. So if you try to understand the scene things by reading the articles about the `getter` and `setter`, you will get the answer.

So you can refactor your code

```rb
  def rent(date, person)
    Rental.new(date, person, self)
  end
```